financial-templates-lib/ExpiringMultiPartyClient.js

```diff
@@ -21,27 +21,22 @@ class ExpiringMultiPartyClient {
   // Returns an array of { sponsor, numTokens, amountCollateral } for each position that is undercollateralized
   // according to the provided `tokenRedemptionValue`.
   getUnderCollateralizedPositions = tokenRedemptionValue => {
-    const { toBN, toWei } = web3.utils;
-    const trv = toBN(tokenRedemptionValue);
-    const fixedPointAdjustment = toBN(toWei("1"));
     return this.positions.filter(position =>
-      // The formula for an undercollateralized position is:
-      // (numTokens * trv) * collateralRequirement > amountCollateral.
-      // Need to adjust by 10**18 twice because each value is represented as a fixed point scaled up by 10**18.
-      toBN(position.numTokens)
-        .mul(trv)
-        .mul(this.collateralRequirement)
-        .gt(
-          toBN(position.amountCollateral)
-            .mul(fixedPointAdjustment)
-            .mul(fixedPointAdjustment)
-        )
+      this._isUnderCollateralized(position.numTokens, position.amountCollateral, tokenRedemptionValue)
     );
   };
 
   // Returns an array of { sponsor, id, numTokens, amountCollateral, liquidationTime } for each undisputed liquidation.
+  // To check whether a liquidation can be disputed, call `isDisputable` with the token redemption value at
+  // `liquidationTime`.
   getUndisputedLiquidations = () => this.undisputedLiquidations;
 
+  // Whether the given `liquidation` (`getUndisputedLiquidations` returns an array of `liquidation`s) is disputable.
+  // `tokenRedemptionValue` should be the redemption value at `liquidation.time`.
+  isDisputable = (liquidation, tokenRedemptionValue) => {
+    return !this._isUnderCollateralized(liquidation.numTokens, liquidation.amountCollateral, tokenRedemptionValue);
+  };
```

Comment:
Should we check here or in the intake step of `_update()` that the liquidation isn't past its expiry/liveness period?

Reply:
+ or either has already been disputed or is pending dispute

Reply:
> ++ or either has already been disputed or is pending dispute

I think that's already checked on line 93 where the liquidations are pulled in.

Reply:
Yep, we should definitely check for that. Done in _update, thoughts?